### ISSUE TITLE
Multiple albums with identical names are not handled properly

### ISSUE THREAD

**Issue by joerghansen** [ampache/ampache#1613]
# Environment

 * Ampache version: 3.8.3
 * Web server + version: nginx 1.12.1
 * Server operating system: Arch Linux
 * Client operating system: Arch Linux
 * Client type:
   - Web: Chromium

# Description

I have two albums with the identical name '25' in my collection (from Adele and a-ha). The file tree looks like (all mp3 files are fully id3v2 tagged):

* mp3
  * Adele
    * 25
      * track01.mp3
      * ...
  * a-ha
    * 25
      * track01.mp3
      * ...

When I add this tree to a catalog it seems that the second album "overwrites" the first one in the database (only one entry with name '25' in the table 'album'). Note that the songs of both albums are added to the table 'songs'.

# Settings
[ampache.cfg.php.txt](https://github.com/ampache/ampache/files/1347342/ampache.cfg.php.txt)

**Comment by lachlan-00** [comment]
That doesn't seem right as the albums are built from the song.

What do you get from this query?
~~~
SELECT * FROM song
LEFT JOIN artist on artist.id = song.artist
LEFT JOIN album on album.id = song.album
WHERE album.name = '25'
~~~

**Comment by joerghansen** [comment]
This query got me the following result:
[select.txt](https://github.com/ampache/ampache/files/1376124/select.txt)
This query contains all tracks from both albums but the album id is identical for all tracks.

**Comment by douglasriper** [comment]
I noticed the same issue with the albums of Morricone and Deep Purpple called "The Platinum Collection".
On up To date dev branch of Ampache.

**Comment by lachlan-00** [comment]
The first problem is your tags are really light/missing information.

The only tag data you have for album is name and year so they match.

At a minimum if you add an album artist it will separate them. You should be doing this anyway for Various and multi-artist albums as well.

**Comment by joerghansen** [comment]
I added the TPE2 tag (album artist) to all my mp3 files and imported these files into an empty Ampache database. I can confirm that the issue is now gone i.e. the album list now contains two entries for both albums '25' from a-ha and Adele.

I understand that the album artist is necessary to separate the albums if the album list is built from the song table. Additionally, I found the corresponding note about the album artist in the FAQ. So, from my point of view, this issue is resolved, and I close it.
